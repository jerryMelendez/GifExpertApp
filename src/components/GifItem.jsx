/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */

export const GifItem = ({title, url, id}) => {
  return (
    <div className="card">
        <img src={url} alt="" />
        <p>{title}</p>
    </div>
  )
}
