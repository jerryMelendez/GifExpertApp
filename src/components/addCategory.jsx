/* eslint-disable react/prop-types */
/* eslint-disable no-unused-vars */
import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = (/*{setCategories}*/ {onNewCategory}) => {
    // eslint-disable-next-line react/prop-types
    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const onSubmit = (event) => {

        // evitar que se recargue la pagina
        event.preventDefault();

        // evitar que un input vacio se agregue al arreglo
        if(inputValue.trim().length < 1) {
            return;
        }

        onNewCategory(inputValue);
        // Setear los valores de las categorias desde el componente hijo al componente padre
        // props.setCategories([inputValue, ...props.categories]);
        // setCategories(cat => [inputValue, ...cat]);

        setInputValue('');
        
    }

    return (
        <form onSubmit={event => onSubmit(event)}>
            <input type="text" 
            placeholder="Buscar Gif" 
            value={inputValue}
            onChange={event => onInputChange(event)}
            />
        </form>
        
    )
}

AddCategory.propTypes = { onNewCategory: PropTypes.func.isRequired };