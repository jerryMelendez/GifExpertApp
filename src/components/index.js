export * from './addCategory.jsx';
export * from './GifGrid.jsx';
export * from './GifItem.jsx';