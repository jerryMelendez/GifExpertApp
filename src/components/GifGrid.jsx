/* eslint-disable no-undef */
/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
import { getGifs } from "../helpers/GetGifs";
import { useEffect, useState } from "react";
import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

// const getGifs = async (category) => {
//   const url = `https://api.giphy.com/v1/gifs/search?api_key=bOyeLvleYcA0xdhlEzQa7OUCR1eTAf7S&q=${encodeURI(category)}&limit=20`;

//   const resp = await fetch(url);

//   const {data} = await resp.json();

//   const gifs = data.map(({id, title, images}) => ({
//     id,
//     title,
//     url: images?.downsized_medium.url
//   }));

//   console.log(gifs);
// }

export const GifGrid = ({category}) => {
  
  const { images, isLoading } = useFetchGifs(category);

  console.log({images, isLoading});

  return (
    <>
        <h3>{category}</h3>

        {
          // Formas de operadores logicos: Si isLoading es true va a renderizar en h2
          // isLoading ? ( <h2>Cargando...</h2> ) : null
          isLoading && (<h2>Cargando...</h2>)
        }

        <div className="card-grid">
          {
              images.map((image) => (
                  <GifItem key={image.id} {... image} />
              ))
          }
        </div>
    </>
  )
}
