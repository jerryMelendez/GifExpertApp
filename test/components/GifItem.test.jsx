/* eslint-disable react/jsx-no-undef */
/* eslint-disable no-undef */

import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";
import { screen } from '@testing-library/react'


describe('Pruebas en el componente', () => {

    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';   

    test('debe de hacer match con el snapshot', () => {
        // render(<GifItem title={title} url={url} />);
        const {container} = render(<GifItem url = {url} title={title}/>);
        
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar la imagen con el url y el alt indicados', () => {
        render(<GifItem url = {url} title={title}/>);
        
        // console.log(screen.getByRole('img').src);
        // console.log(screen.getByRole('img').alt);

        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const {src, alt} = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('debe mostrar un titulo en el componente', () => {
        render(<GifItem url = {url} title={title}/>);

        // Comprobar que existe un elemento con el texto indicado en el componente renderizado
        expect(screen.getByText(title)).toBeTruthy();
    }
    );
});