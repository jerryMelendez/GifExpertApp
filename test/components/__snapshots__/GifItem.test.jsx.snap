// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Un titulo"
      src="https://localhost/algo.jpg"
    />
    <p>
      Un titulo
    </p>
  </div>
</div>
`;
